### PR TITLE
feat: Add delete multisig account option

### LINKED ISSUE #743: multisig: Allow user to delete existing multisig accounts
- [ ] Add a new REST endpoint which accept multisig account address and creator address. If the creator address is valid first remove all the transactions related to given multisig account and after that delete multisig account.
 - [ ] In frontend Inform user before deleting the multisig account.

### PR DESCRIPTION
**Pull request by Hemanthghs**
Closes #743

### CODE REVIEW

File: frontend/src/components/multisig/DialogConfirm.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  const { open, deleteMultisigAccount, onClose } = props;
+
+  return (
+    <div>
+      <Dialog onClose={onClose} open={open}>
+        <DialogContent
+          sx={{ textAlign: "center", fontWeight: 600 }}
+          variant="h6"
+        >
+          Delete multisig account?
```
Comment (aleem1314) [inline frontend/src/components/multisig/DialogConfirm.jsx:20]:
```suggestion
          Deleting a multisig account removes all its transactions.
```

File: server/handler/transactions.go [hunk 2]
```diff
@@ -380,3 +380,74 @@ func (h *Handler) DeleteTransaction(c echo.Context) error {
...
+			})
+		}
+		return c.JSON(http.StatusInternalServerError, model.ErrorResponse{
+			Status:  "error",
+			Message: "failed to get accounts",
+			Log:     err.Error(),
+		})
+	}
+
+	_, err := h.DB.Exec(`DELETE from transactions WHERE multisig_address=$1`, address)
```
Comment (aleem1314) [inline server/handler/transactions.go:420]: We should use transaction for all database operations. If any action fails, we should undo them.

File: server/server.go [hunk 3]
```diff
@@ -68,6 +68,7 @@ func main() {
 	e.DELETE("/multisig/:address/tx/:id", h.DeleteTransaction)
 	e.POST("/multisig/:address/sign-tx/:id", h.SignTransaction)
 	e.GET("/multisig/:address/txs", h.GetTransactions)
+	e.DELETE("/multisig/:creator-address/:address", h.DeleteMultisigAccount)
```
Comment (aleem1314) [inline server/server.go:71]: This API allows anyone to delete multisig accounts if they know the creator's address. @anilCSE are we okay with this one.

Comment (anilCSE) [reply server/server.go:71]: Oh that's bad. We can instead introduce archive mode may be. Also, let's restrict from hitting this API from direct calls.

Comment (Hemanthghs) [reply server/server.go:71]: For restricting unauthorized access to delete multisig account api, We can use keplr's signArbitrary API for user verification.
signArbitrary takes chainID, address, and any data string. If user approves that, it is returning a signature.
We can store that signature in localstorage and hash of signature in our database.
For delete multisig account api we can send the signature from localStorage as bearer token in payload and verify the signature with hash stored in DB before we delete account data in our db.
Can we use this?
https://docs.keplr.app/api/#request-signature-for-arbitrary-message

image

**Pull request closed without merging**